src/Filter.Nest/QueryContainerDescriptorExtensions.cs

```diff
@@ -0,0 +1,99 @@
+﻿using Nest;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace RimDev.Filter.Nest
+{
+    public static class QueryContainerDescriptorExtensions
+    {
+        public static QueryContainer Filter<T>(
+            this QueryContainerDescriptor<T> value,
+            object filter,
+            IDictionary<Type, Func<object, string>> filterValueFormatters = null)
+            where T : class
+        {
+            if (filter == null)
```

Comment:
Nice.

![Kylo Ren understands the meaning of teamwork](https://media.tenor.co/images/dd58d550e99e2338dddb4bdfb4cf65f3/raw)